src/Workspaces/Remote/ServiceHub/Services/ConvertTupleToStructCodeRefactoringProvider/RemoteConvertTupleToStructCodeRefactoringService.cs

```diff
@@ -17,19 +20,47 @@ namespace Microsoft.CodeAnalysis.Remote
 {
     internal sealed class RemoteConvertTupleToStructCodeRefactoringService : BrokeredServiceBase, IRemoteConvertTupleToStructCodeRefactoringService
     {
-        internal sealed class Factory : FactoryBase<IRemoteConvertTupleToStructCodeRefactoringService>
+        internal sealed class Factory : FactoryBase<IRemoteConvertTupleToStructCodeRefactoringService, IRemoteConvertTupleToStructCodeRefactoringService.ICallback>
         {
-            protected override IRemoteConvertTupleToStructCodeRefactoringService CreateService(in ServiceConstructionArguments arguments)
-                => new RemoteConvertTupleToStructCodeRefactoringService(arguments);
+            protected override IRemoteConvertTupleToStructCodeRefactoringService CreateService(in ServiceConstructionArguments arguments, RemoteCallback<IRemoteConvertTupleToStructCodeRefactoringService.ICallback> callback)
+                => new RemoteConvertTupleToStructCodeRefactoringService(arguments, callback);
         }
 
-        public RemoteConvertTupleToStructCodeRefactoringService(in ServiceConstructionArguments arguments)
+        private readonly RemoteCallback<IRemoteConvertTupleToStructCodeRefactoringService.ICallback> _callback;
+
+        public RemoteConvertTupleToStructCodeRefactoringService(in ServiceConstructionArguments arguments, RemoteCallback<IRemoteConvertTupleToStructCodeRefactoringService.ICallback> callback)
             : base(arguments)
         {
+            _callback = callback;
         }
 
+#if TODO // Replace the below specialization with a call to a generic impl once https://github.com/microsoft/vs-streamjsonrpc/issues/789 is fixed
```

Comment:
Creates options provider that fetches options from the client and caches them for a given language.